ls1intum/Artemis
# [Refactor] Refactoring of the saving process in exercise-update-component

### Checklist

### Motivation and Context
After working with `exercise-update-component.ts` files, I have noticed some duplication. One of them revolves around save method. This PR moves the save logic to a separate class and calls it from these components, removing duplication.

### Description

I introduce a new class called `SaveExerciseCommand<T extends Exercise>` which performs the save for the specified `Exercise` as well as a `ExerciseServicable<T extends Exercise>` that is an interface that other ExerciseServices conform to.

### Steps for testing

1. Log in to Artemis
2. Navigate to Course Management
3. Pick a course
4. Create new text-/modeling-/file-upload exercises
5. Creating/Updating the exercises should work as in develop

### Test Coverage
Exercise-Utils: 94.95%
